Zep: Authentication; other bugfixes/updates; bump SDK version

danielchalef opened the pull request:
- zep auth; auth in example; fix for lastn
- if summary exists, add it to message history
- update class names
- bump zep-js version

rsharath commented:
@nfcampos / @jacoblee93  - this change is ready for merge, appreciate you taking a look. Thanks!

jacoblee93 commented on langchain/src/memory/zep.ts:
```diff
@@ -51,15 +55,17 @@ export class ZepMemory extends BaseChatMemory implements ZepMemoryInput {
 
   async loadMemoryVariables(values: InputValues): Promise<MemoryVariables> {
-    const lastN = values.lastN ?? 10;
+    // use either lastN provided by developer or undefined to use the
```
Will this default to 10 for current users?

danielchalef replied:
This is a bug fix. The number of results returned should default to the message window size set on the server, not 10.

jacoblee93 commented on langchain/package.json:
```diff
@@ -445,7 +445,7 @@
     "@faker-js/faker": "^7.6.0",
     "@getmetal/metal-sdk": "^4.0.0",
-    "@getzep/zep-js": "^0.3.1",
+    "@getzep/zep-js": "^0.4.1",
```
Update the peer dep as well

danielchalef replied:
Fixed

jacoblee93 reviewed (changes requested):
The integration test is now failing for me - can you fix?

danielchalef commented:
This is a bug fix. The number of results returned should default to the message window size set on the server, not 10.

I’ll take a look at why the integration test is failing.

On Wed, Jun 28 2023 at 6:12 PM, Jacob Lee < ***@***.*** > wrote:

> 
> 
> 
> 
> 
> 
> 
> ***@***.**** commented on this pull request.
> 
> 
> 
> In langchain/src/memory/zep.ts (
> https://github.com/hwchase17/langchainjs/pull/1766#discussion_r1245976670
> ) :
> 
> >    }
>  
>    get memoryKeys() {
>      return [this.memoryKey];
>    }
>  
>   
> async loadMemoryVariables(values: InputValues): Promise<MemoryVariables> {
> -    const lastN = values.lastN ?? 10;
> +    // use either lastN provided
> by developer or undefined to use the
> 
> 
> Will this default to 10 for current users?
> 
> 
> 
> —
> Reply to this email directly, view it on GitHub (
> https://github.com/hwchase17/langchainjs/pull/1766#pullrequestreview-1504425674
> ) , or unsubscribe (
> https://github.com/notifications/unsubscribe-auth/AABAAZZQWB25TGRLLBGCDCLXNTI67ANCNFSM6AAAAAAZUZTSDU
> ).
> You are receiving this because you authored the thread. Message ID: <hwchase17/langchainjs/pull/1766/review/1504425674
> @ github. com>
> 
> 
>

danielchalef commented:
@jacoblee93 I've updated the peer dep, improved the robustness of the retriever, and added an additional test. Tests pass locally. Can you let me know if the CI tests pass? I can't see the results here.

As an aside, I have to manually modify the `test.skip` to `test` in to run them using `yarn test:single`. Is there a better way to force them to run from the CLI?

jacoblee93 commented:
> @jacoblee93 I've updated the peer dep, improved the robustness of the retriever, and added an additional test. Tests pass locally. Can you let me know if the CI tests pass? I can't see the results here.
> 
> As an aside, I have to manually modify the `test.skip` to `test` in to run them using `yarn test:single`. Is there a better way to force them to run from the CLI?

The alternative is not skipping them at all by adding a container setup here:
https://github.com/hwchase17/langchainjs/blob/main/test-int-deps-docker-compose.yml

jacoblee93 commented:
Fixed out of date yarn.lock and merged in with #1811

danielchalef commented:
Thanks, @jacoblee93 !
